src/NumericalAlgorithms/Interpolation/Callbacks/FindApparentHorizon.hpp

```diff
@@ -0,0 +1,171 @@
+/// struct called `post_horizon_find_callback`, which has a function
+///```
+///  static void apply(const DataBox<DbTagsConst>&,
+///                    const intrp::ConstGlobalCache<Metavariables>&,
+///                    const Metavariables::temporal_id&) noexcept;
+///```
+/// that is called if the FastFlow iteration has converged.
+///
+/// Uses:
+/// - Metavariables:
+///   - `temporal_id`
+///   - `domain_frame`
+/// - DataBox:
+///   - `::Tags::Verbosity`
+///   - `::gr::Tags::InverseSpatialMetric<3,Frame>`
+///   - `::gr::Tags::ExtrinsicCurvature<3,Frame>`
+///   - `::gr::Tags::SpatialChristoffelSecondKind<3,Frame>`
+///   - `::Tags::FastFlow`
+///   - `StrahlkorperTags::Strahlkorper<Frame>`
+///
+/// Modifies:
+/// - DataBox:
+///   - `::Tags::FastFlow`
+///   - `StrahlkorperTags::Strahlkorper<Frame>`
+///
+/// This is an InterpolationTargetTag::post_interpolation_callback;
+/// see InterpolationTarget for a description of InterpolationTargetTag.
+template <typename InterpolationTargetTag>
+struct FindApparentHorizon {
+  template <typename DbTags, typename DbTagsConst, typename Metavariables>
+  static bool apply(
+      const gsl::not_null<db::DataBox<DbTags>*> box,
+      const db::DataBox<DbTagsConst>& const_box,
+      const gsl::not_null<Parallel::ConstGlobalCache<Metavariables>*> cache,
+      const typename Metavariables::temporal_id& temporal_id) noexcept {
+    const auto& verbosity = db::get<::Tags::Verbosity>(const_box);
+    const auto& inv_g = db::get<::gr::Tags::InverseSpatialMetric<
+        3, typename Metavariables::domain_frame>>(const_box);
+    const auto& ex_curv = db::get<::gr::Tags::ExtrinsicCurvature<
+        3, typename Metavariables::domain_frame>>(const_box);
+    const auto& christ = db::get<::gr::Tags::SpatialChristoffelSecondKind<
+        3, typename Metavariables::domain_frame>>(const_box);
+
+    std::pair<FastFlow::Status, FastFlow::IterInfo> status_and_info;
+
+    // Do a fastflow iteration.
+    db::mutate<::Tags::FastFlow, StrahlkorperTags::Strahlkorper<
+                                     typename Metavariables::domain_frame>>(
+        box, [&inv_g, &ex_curv, &christ, &status_and_info ](
+                 const gsl::not_null<::FastFlow*> fast_flow,
+                 const gsl::not_null<
+                     ::Strahlkorper<typename Metavariables::domain_frame>*>
+                     strahlkorper) noexcept {
+          status_and_info = fast_flow->template iterate_horizon_finder<
+              typename Metavariables::domain_frame>(strahlkorper, inv_g,
+                                                    ex_curv, christ);
+        });
+
+    // Determine whether we have converged, whether we need another step,
+    // or whether we have encountered an error.
+
+    const auto& status = status_and_info.first;
+    const auto& info = status_and_info.second;
+
+    if (verbosity > ::Verbosity::Quiet or
+        (verbosity > ::Verbosity::Silent and converged(status))) {
+      Parallel::printf(
```

Comment:
It would be nice to print some context for the following message (e.g.  "Apparent horizon finder information: \n" and define what R |R| |R_grid| and r are)